src/org/folio/rest_v2/Eureka.groovy

```diff
@@ -0,0 +1,95 @@
+package org.folio.rest_v2
+
+import org.folio.models.EurekaTenant
+import org.folio.utilities.RequestException
+
+class Eureka extends Authorization {
+
+  /**
+   * EurekaTenant object contains Master Tenant configuration for Eureka.
+   * @attribute 'tenantId'     Default: "master"
+   * @attribute 'clientId'     Default: "folio-backend-admin-client"
+   */
+  public EurekaTenant masterTenant
+
+  Eureka(Object context, String eurekaDomain, EurekaTenant masterTenant, boolean debug = false) {
+    super(context, eurekaDomain, debug)
+    this.masterTenant = masterTenant
+  }
+
+  void createTenant(EurekaTenant tenant) {
+    if (isTenantExist(tenant.tenantId)) {
+      logger.warning("Tenant ${tenant.tenantId} already exists!")
+      return
+    }
+
+    String url = generateUrl("/tenants")  // Tenant Manager URL
+    Map<String, String> headers = getMasterHeaders()
+    Map body = [
+      name: tenant.tenantId,
+      description: tenant.tenantDescription
+    ]
+
+    logger.info("Creating tenant ${tenant.tenantId}...")
+
+    restClient.post(url, body, headers)
+
+    logger.info("Tenant (${tenant.tenantId}) successfully created")
+  }
+
+  boolean isTenantExist(String tenantId) {
+    String url = generateUrl("/tenants/${tenantId}")
+    Map<String, String> headers = getMasterHeaders()
+
+    try {
+      restClient.get(url, headers)
+      logger.info("Tenant ${tenantId} exists")
+      return true
+    } catch (RequestException e) {
+      if (e.statusCode == HttpURLConnection.HTTP_NOT_FOUND) {
```

Comment:
Why don't we extract status from rest client itself?

Reply:
@eldiiar-duishenaliev 

Method **doRequest** doesn't return Http Response Code:
https://github.com/folio-org/pipelines-shared-library/blob/bbfbd459f27514f51d005ba2efb5492b2dfd45fe/src/org/folio/utilities/RestClient.groovy#L68

Should I develop **doRequest_v2** method that returns Http Response Code?

Reply:
At the moment there is only one way to get responce code from RestClient object:

Analyze **RequestException.statusCode** value

Reply:
no, we don't need any check in catch block, since our initial attempt already in fail state(if any).

Reply:
@eldiiar-duishenaliev 
Should I develop **doRequest_v2** method that returns Http Response Code?

Reply:
Code is changed accordingly to comment.